### PR TITLE
test: include sandbox in canary tests

### PR DESCRIPTION
**Pull request by sobolk**
*Issue #, if available:*

*Description of changes:*

This change adds sandbox to canary test suite.

There's one workaround applied to locate amplify binary in local installation. Better solution will be pursued here https://github.com/aws-amplify/amplify-backend/issues/582 .

Canary run from branch https://github.com/aws-amplify/amplify-backend/actions/runs/6816085805 .

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/integration-tests/src/test-live-dependency-health-checks/health_checks.test.ts [hunk 1]
```diff
@@ -35,50 +38,91 @@ void describe('Live dependency health checks', () => {
     }
   });
 
-  beforeEach(async () => {
-    tempDir = await fs.mkdtemp(path.join(os.tmpdir(), 'test-amplify'));
-    testBranch = await amplifyAppPool.createTestBranch();
-  });
+  void describe('pipeline deployment', () => {
+    let tempDir: string;
+    let testBranch: TestBranch;
```
Comment (sobolk) [inline packages/integration-tests/src/test-live-dependency-health-checks/health_checks.test.ts:43]: enabling parallelization requires nesting of these variables

**Review by Amplifiyer — DISMISSED** [review]

File: packages/integration-tests/src/process-controller/process_controller.ts [hunk 2]
```diff
@@ -123,8 +123,16 @@ export const amplifyCli = (
 ): ProcessController => {
   let command: string;
   if (options?.installationType === 'local') {
```
Comment (edwardfoyle) [inline packages/integration-tests/src/process-controller/process_controller.ts:125]: Could we get rid of this if statement and do the lookup unconditionally?

File: packages/integration-tests/src/process-controller/process_controller.ts [hunk 3]
```diff
@@ -123,8 +123,16 @@ export const amplifyCli = (
...
-    command = 'npx';
-    args = ['amplify'].concat(args);
+    // TODO This is a workaround to lookup locally installed binary as seen by npx
+    // We're using binary directly because signals (Ctrl+C) don't propagate
+    // to child processes without TTY emulator.
+    // See: https://github.com/aws-amplify/amplify-backend/issues/582
+    command = execaSync('npx', ['which', 'amplify'], {
+      cwd: dir,
+    }).stdout;
```
Comment (edwardfoyle) [inline packages/integration-tests/src/process-controller/process_controller.ts:132]: do we need a `.trim()` here?

**Review by edwardfoyle — APPROVED** [review]
changing to approve. I don't think my comments are blockers

File: packages/integration-tests/src/process-controller/process_controller.ts (hunk 2 above)
Comment (sobolk) [reply packages/integration-tests/src/process-controller/process_controller.ts:125]: easy to try.
If it works it works. if not we'll keep this.

**Review by Amplifiyer — APPROVED** [review]

File: packages/integration-tests/src/process-controller/process_controller.ts (hunk 2 above)
Comment (sobolk) [reply packages/integration-tests/src/process-controller/process_controller.ts]: Tests are happy, we can do it.

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**